docs/mp/oci/01_OciObjectStorage.adoc

```diff
@@ -0,0 +1,205 @@
+        </dependency>
+----
+
+
+== Setting up the Object Storage
+
+
+In order to use the OCI Object Storage integration, the following setup should be made:
+
+Current configuration requires `~/.oci/config` to be available in the home folder. This configuration file can be downloaded from OCI.
+
+REST endpoint to work with OCI Object Storage:
+
+[source,java]
+----
+@Path("/files")
+public class ObjectStorageResource {
+    private final OciObjectStorage objectStorage;
+    private final String bucketName;
+
+    @Inject
+    ObjectStorageResource(OciObjectStorage objectStorage, (1)
+                          @ConfigProperty(name = "oci.objectstorage.bucket")
+                                  String bucketName) { (2)
+        this.objectStorage = objectStorage;
+        this.bucketName = bucketName;
+    }
+}
+----
+1. `OciObjectStorage` is configured and injected automatically
+2. Bucket name is read from the properties
+
+Additionally, in `application.yaml` OCI properties should be specified:
+
+[source,yaml]
+----
+oci:
+    properties:
+      compartment-ocid: "ocid1.tenancy.oc1..<..>"
+      objectstorage-namespace: "<..>"
+      objectstorage-bucket: "demobucket"
+
+----
+
+The exact values are available in OCI object storage and bucket properties.
+
+== Using the Object Storage
+
+=== Upload file
+To upload a file via `PUT` method to OCI Object Storage:
+
+[source,java]
+----
+    @POST
+    @Path("/file/{fileName}")
+    public Response upload(@PathParam("fileName") String fileName,
+                         @HeaderParam("Content-Length") long contentLength,
+                         @HeaderParam("Content-Type") @DefaultValue("application/octet-stream") String type,
+                         InputStream entity) {
+        PutObject.Response response = objectStorage.putObject(PutObject.Request.builder() (1)
+                                                                      .contentLength(contentLength)
+                                                                      .bucket(bucketName)
+                                                                      .requestMediaType(io.helidon.common.http.MediaType
+                                                                                                .parse(type))
+                                                                      .objectName(fileName),
+                                                              Channels.newChannel(entity));
+
+        return Response.status(response.status().code()) (2)
+                .header("opc-request-id", response.headers().first("opc-request-id").orElse(""))
+                .header("request-id", response.requestId())
+                .build();
+    }
+----
+
+1. Use `objectStorage.putObject` method with`PutObject.Request.builder()` to submit data
+3. Put the result in the `Response`
+
+
+=== Download file
+To download the file via `GET` method from OCI Object Storage:
+
```

Comment:
It would be good to reformat the code so it is rendered nicely (for all methods) - the output seems to be quite bad for the current formatting.

Reply:
fixed